### DISCUSSION TITLE
what do you think about design pattern

### DISCUSSION THREAD

**Discussion by blotterchains** [Godot-ground/tower-defense#1] [Q&A] [answered]
Please, after a few hours of playing, express your opinions regarding the initial design of the game and how to make files and nodes from your perspective(artist,developer,game designer)

**Comment by danialabdl** [comment]
so we need units in game with healthbar and ability to shoot once awhile and enemies with healthbar and walking or running ability.
we need a wave manager for waves of enemies in each level.

**Comment by blotterchains** [comment]
this is my idea for project structure.

in Scenes we must combine 2 part of Arts and Codes parts
and art and dev teams just works on these folders and at the end level designer combine that into level
and a person who combine arts and codes into Components(mayber another developer who know engine very well).components can be a Button,health bar,animatioans and UI combined with code
and in general we got other things like building or anything like that.Enemies and Heros can have subdirectory like guns or level ups

and at the end level designer works with Levels folder and combine scenes into it.
i think developers must combine other parts of this folder with codes and arts.and after make it work level designer make it into a level.
and from this point we use Wave instead of level.please mention your idea what make it wrong in large scale

Project_ROOT
=============== Scenes
=============== ========= Components
=============== ========= Levels
=============== ========= General
=============== ========= Enemies
=============== ========= Heros
=============== Arts
=============== ========= Sounds
=============== ========= Sprites
=============== ========= General
=============== ========= UI
=============== Codes
=============== ========= Components
=============== ========= Levels
=============== ========= General
=============== ========= Enemies
=============== ========= Heros

**Comment by danialabdl** [comment, marked as answer]
i think "components" and "general" are overlapping. maybe components is not neccessary. we have scenes in those combine thing together.
my suggestion
under scene folder we can have :
-levels
-heroes
-enemies
-menus(main, setting, shpping)
-general

under code:
-level
-ui
-heroes
-enemies
-general
art:
-ui
-heroes
-enemies
-levels
-resources(fonts, sounds, tiles...)

**Comment by blotterchains** [comment]
ok.i marked it as answer.
if you guys are new to godot-ground and think we are in wrong state of design please reopen this

**Comment by Alinj7** [comment]
We always need to think of new things for the progress of our project. 
 A scenario about soldiers fighting is a good scenario but not the best scenario. 
 This scenario is repetitive and overused.  But if, for example, instead of soldiers, we use Indians and archers, now we have a new scenario.  
There are many other examples that do not change the coding.  For example, the battle of Greek legends instead of soldiers!
